app/api/api_v1/endpoints/job_router.py

```diff
@@ -0,0 +1,39 @@
+from typing import List
+from pydantic import BaseModel
+from uuid import UUID
+from fastapi import APIRouter, Request, Depends, UploadFile, File
+from sqlalchemy.orm import Session
+from app.celery import get_task_result_by_id
+from app.celery.tasks import downsync_task
+from app.services import LmsSyncService, AssignmentService
+from app.schemas import JobSchema, JobResultSchema
+from app.core.dependencies import (
+    get_db, PermissionDependency,
+    UserIsInstructorPermission
+)
+
+router = APIRouter()
+
+@router.get("/jobs/{id}", response_model=JobResultSchema)
+async def get_job(
+    *,
+    id: UUID,
+    perm: None = Depends(PermissionDependency(UserIsInstructorPermission))
+):
+    """
+    NOTE: Task ID is not validated. A pending task does not necessarily exist;
+    this state is returned by default for unknown task ids.
```

Comment:
What does "this state" mean exactly?

Reply:
Pending state

Reply:
I'm assuming that there's some underlying reason why this is the way it is, like Celery itself returns `Pending` for jobs that don't exist? If so, is it important to know when a task doesn't exist, or do we not really care if it exists or not if we query a job id and get `Pending` in most cases? I feel like this method should either check if the job exists before everything and return a different state if not, or if it gets a `Pending` state back from `JobResultSchema.from_async_result(get_task_result_by_id(str(id)))` it should then check to make sure the job exists.

Reply:
Yes, that's what the comment is trying to convey. It's not important to know (and not particularly easy to do so using Celery directly without inspecting the broker store which I'm not convinced is even a consistent solution and is likely still somewhat ambiguous), nor should the method be doing so due to the complexity it introduces, without marked benefit. A non-existent task is implied to be pending, it's just the default state.